.ci/ci_cache_components.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+cidir=$(dirname "$0")
+source "${cidir}/../lib/common.bash"
+source "${cidir}/lib.sh"
+WORKSPACE=${WORKSPACE:-$(pwd)}
+kata_dir="/usr/share/kata-containers"
+
+cache_qemu_artifacts() {
+	current_qemu_version=$(get_version "assets.hypervisor.qemu.version")
+	qemu_tar="kata-qemu-static.tar.gz"
+	build_cache "$qemu_tar" "${current_qemu_version}"
+}
+
+cache_kernel_artifacts() {
+	for k in "${kata_dir}"/*.container; do
+		echo "Adding ${k}"
+		real_path=$(readlink -f "${k}")
+		#Get version from binary (format: vmlinu{z|x}-4.19.24-25)
+		kernel_binary=$(basename "${real_path}")
+		current_kernel_version=$(echo "${kernel_binary}" | cut -d- -f2-)
+		build_cache "${k}" "${current_kernel_version}"
+	done
+}
+
+cache_image_artifacts() {
+	image="${kata_dir}/kata-containers.img"
+	path=$(readlink -f "${image}")
+	image_version=$(echo $(basename "${path}"))
+	build_cache "${image}" "${image_version}"
+	create_image_tar "${image}"
+}
+
+cache_nemu_artifacts() {
+	nemu_version=$(get_version "assets.hypervisor.nemu.version")
+	nemu_tar="kata-nemu-static.tar.gz"
+	build_cache "${nemu_tar}" "${nemu_version}"
+}
+
+cache_image_initrd_artifacts() {
+	image="${kata_dir}/kata-containers-initrd.img"
+	path=$(readlink -f "${image}")
+	image_version=$(echo $(basename "${path}"))
+	build_cache "${image}" "${image_version}"
+	create_image_tar "${image}"
+}
+
+create_image_tar(){
+	image="$1"
+	sha256sum -c "sha256sum-${image}"
+	tar -cJf "$(cat latest-${image}).targ.xz" "$(cat latest-${image})"
+	sha256sum "$(cat latest-${image}).tar.xz" > "sha256sum-${image}-tarball"
```

Comment:
Conventionally, checksum files tend to be named `${file}.${checksum_type}` so I was expecting this to be `foo.tar.xz.sha256[sum]` (or `foo.tar.xz.sha512[sum]`). That way, the files sort sanely with `ls` too.